docs/README_DOCS.md

```diff
@@ -0,0 +1,127 @@
+
+# Welcome to the mewbot docs!
+
+Mewbot is a plugin friendly, multi-platform, chatbot framework which can load bots from YAML definitions.
+
+The docs are built using `sphinx`.
+
+# What should go in here?
+
+Anything and everything related to mewbot
+ - installing it 
+ - running it
+ - debugging it
+
+This should be the primary knowledge repository for mewbot.
+(Things like github issues are all well and good, but github itself may go away one day.)
+
+# Why are there a mix of md files and rst files in here?
+
+`sphinx`, by default, uses`.rst` files.
+However it can use `.md` files, and others, with appropriate plugins.
+Some of the files are legacy files which where written in a certain format.
+As such they have not, yet, been translated.
+
+# How do I build this documentation?
+
+## If you want to build it manually (NOT RECOMMENDED)
+
+1) Make sure you've run `pip install -r requirements-dev.txt` sometime recently.
+2) This will install `sphinx` and the plugins it needs to build
+3) `cd` into `./docs` (the folder containing this file)
+4) Run `sphinx-apidoc -o source_mewbot/ ../src/mewbot` - This will recursively generate sphinx docs for the entire code base off the docstrings embedded in the code itself.
+5) Run `sphinx-apidoc -o source_examples/ ../src/examples` - This will do the same for the examples .
+6) The output of this will be stored in the `source_mewbot` and `source_examples` folders respectively.
+7) NOTE - doc strings in the source are considered a source of truth. Please DO NOT edit the generated files - please edit the doc strings in `src` directly.
+8) Run `python generate_sub_indexes.py` - this will generate index files for `design-docs`, `dev-docs` e.t.c. as well as the auto generated fies in `source_mewbot` and `source_examples`
+9) Delete `_build` - to actually make these changes show up anywhere.
+10) Run `make html` (or any of the other output methods supported by `sphinx` - for a full list see their documentation)
+11) The doc files will be generated in the `_build/html` in this directory
+12) Open `_build\html\index.html` with your favourite web browser, and enjoy!
+
+## If you want to use mewbot's internal tooling (RECOMMENDED)
+
+1) Make sure you've run `pip install -r requirements-dev.txt` sometime recently.
+2) make sure you have your mewbot `venv` activated
+3) Run `python -m mewbot.tools.docs` to build
+
+# How to I contribute to these docs?
+
+More documentation is always welcome!
+These docs are intended as a living document.
+Feel free to add to them.
+In particular, details of problems you encountered - and how you solved them - would be most welcome.
```

Comment:
Probably worth linking back to contributing?

Reply:
Additionally, are we setting an explicit preference to `.rst` files that should be mentioned here? It's sort of implied by the "why is this a mix?" question.

Reply:
Link added.

... I was hoping now to (allowing people to use whatever format they feel more comfortable with), but the translation layer used to incorporate .md into sphinx is a lot less than perfect. So it's probbly worth stating the preference.